src/core/Form/Combobox/Combobox/Combobox.test.tsx

```diff
@@ -0,0 +1,364 @@
+import React from 'react';
+import { render, fireEvent, act } from '@testing-library/react';
+import { axeTest } from '../../../../utils/test/axe';
+import { Combobox } from './Combobox';
+
+const tools = [
+  {
+    name: 'Jackhammer',
+    price: 230,
+    tax: false,
+    labelText: 'Jackhammer',
+    uniqueItemId: 'jh2435626',
+  },
+  {
+    name: 'Hammer',
+    price: 15,
+    tax: true,
+    labelText: 'Hammer',
+    uniqueItemId: 'h9823523',
+  },
+  {
+    name: 'Sledgehammer',
+    price: 36,
+    tax: false,
+    labelText: 'Sledgehammer',
+    uniqueItemId: 'sh908293482',
+  },
+  {
+    name: 'Spade',
+    price: 50,
+    tax: true,
+    labelText: 'Spade',
+    uniqueItemId: 's82502335',
+  },
+  {
+    name: 'Powersaw',
+    price: 150,
+    tax: false,
+    labelText: 'Powersaw',
+    disabled: true,
+    uniqueItemId: 'ps9081231',
+  },
+  {
+    name: 'Shovel',
+    price: 115,
+    tax: true,
+    labelText: 'Shovel',
+    uniqueItemId: 's05111511',
+  },
+  {
+    name: 'Iron stick',
+    price: 85,
+    tax: false,
+    labelText: 'Iron stick',
+    uniqueItemId: 'is3451261',
+  },
+  {
+    name: 'Rake',
+    price: 50,
+    tax: true,
+    labelText: 'Rake',
+    uniqueItemId: 'r09282626',
+  },
+  {
+    name: 'Motorsaw',
+    price: 450,
+    tax: false,
+    labelText: 'Motorsaw',
+    disabled: true,
+    uniqueItemId: 'ms6126266',
+  },
+];
+
+const defaultSelectedTools = [
+  {
+    name: 'Hammer',
+    price: 15,
+    tax: true,
+    labelText: 'Hammer',
+    uniqueItemId: 'h9823523',
+  },
+  {
+    name: 'Powersaw',
+    price: 150,
+    tax: false,
+    labelText: 'Powersaw',
+    disabled: true,
+    uniqueItemId: 'ps9081231',
+  },
+];
+
+const BasicCombobox = (
+  <Combobox
+    labelText="Combobox"
+    items={tools}
+    chipListVisible={true}
+    chipActionLabel="Remove"
+    removeAllButtonLabel="Remove all selections"
+    visualPlaceholder="Choose your tool(s)"
+    emptyItemsLabel="No items"
+    defaultSelectedItems={defaultSelectedTools}
+  />
+);
+
+it('should not have basic accessibility issues', async () => {
+  await act(async () => {
+    axeTest(BasicCombobox);
+  });
+});
+
+describe('Chips', () => {
+  it('should have selected Chips shown', async () => {
+    await act(async () => {
+      const { container } = render(BasicCombobox);
+      expect(container.querySelectorAll('.fi-chip').length).toEqual(2);
+    });
+  });
+
+  it('first Chip should be removable and removed when clicked', async () => {
+    await act(async () => {
+      const { container } = render(BasicCombobox);
+      let chips = container.querySelectorAll('.fi-chip');
```

Comment:
Would it be possible to follow testing library practices more closely and use getByText?

Reply:
It is possible, but thought was to reuse the `chips` as the length is needed in later part of the test. 
And also at the time of writing the test it was thought that the list items (popover) are visible also in the DOM and therefore causing troubles to find the correct one with minimum effort.

Reply:
Sometimes it is cumbersome to use "usage based" selectors. Especially when there is multiple layers on top of the actual element containing the text.

Reply:
Actually, I would prefer to check with getByText rather than relying on length only. But not a big deal and will leave it up to you to either leave as is or change.

Reply:
Test is now changed to use `getByText` and `queryByText` because getByText throws an error if trying to find an element that does not exist.
Not relying on the **length** anymore